modules/core/src/main/java/org/apache/ignite/Ignition.java

```diff
@@ -538,7 +556,43 @@ public static Ignite ignite(@Nullable String name) throws IgniteIllegalStateExce
      * @throws IllegalArgumentException Thrown if current thread is not an {@link IgniteThread}.
      */
     public static Ignite localIgnite() throws IgniteIllegalStateException, IllegalArgumentException {
-        return IgnitionEx.localIgnite();
+        return sandboxIgniteProxy(IgnitionEx.localIgnite());
+    }
+
+    /**
+     * @param ignite Ignite.
+     * @return Ignite component proxy if the Ignite Sandbox is enabled.
+     */
+    private static Ignite sandboxIgniteProxy(Ignite ignite) {
+        return sandboxIgniteProxy(ignite, true);
+    }
+
+    /**
+     * @param ignite Ignite.
+     * @return Ignite component proxy if the Ignite Sandbox is enabled.
+     */
+    private static Ignite sandboxIgniteProxy(Ignite ignite, boolean doInsideSandboxCheck) {
+        if (ignite instanceof IgniteEx && (!doInsideSandboxCheck || insideSandbox())) {
```

Comment:
Do we really need `IgniteEx` check? I can't imagine in which cases `IgnitionEx` will return something else.